osquery/events/windows/usn_journal_reader.h

```diff
@@ -0,0 +1,279 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <atomic>
+#include <ctime>
+#include <memory>
+#include <ostream>
+#include <sstream>
+#include <vector>
+
+#include <boost/functional/hash.hpp>
+
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
```

Comment:
Is this defined generally within CMake? (I am asking to prompt a follow-up, should we stop checking/defining it source?)

Reply:
> Is this defined generally within CMake? (I am asking to prompt a follow-up, should we stop checking/defining it source?)

I'm not sure -- it _looks_ like it gets defined here, but I haven't done the full tracethrough:

https://github.com/osquery/osquery/blob/76720bea2fbf307d070ece6b35ba0ba6c568dbd1/cmake/flags.cmake#L251

I'll add an `#error` locally to see if it's already defined and drop from the header if so.

Reply:
Hmm, I can't find a corresponding `WIN32_LEAN_AND_MEAN` definition for the Buck build, so it might make sense to keep this around.